quinn/src/endpoint.rs

```diff
@@ -257,12 +258,19 @@ where
     fn drive_recv(&mut self, cx: &mut Context, now: Instant) -> Result<bool, io::Error> {
         let mut recvd = 0;
         loop {
-            match self.socket.poll_recv(cx, &mut self.recv_buf) {
-                Poll::Ready(Ok((n, addr, ecn))) => {
-                    match self
-                        .inner
-                        .handle(now, addr, ecn, (&self.recv_buf[0..n]).into())
-                    {
+            let mut meta = [RecvMeta::default()];
+            match self
```

Comment:
I don't love the multi-line match expression here, how do you feel about naming `&mut [IoSliceMut::new(&mut self.recv_buf)]` to split it up?

Reply:
Maybe this runs into borrowck?

Reply:
Works out just fine, in fact!